src/sdk/main/include/TokenRejectFlow.h

```diff
@@ -0,0 +1,217 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef HEDERA_SDK_CPP_TOKEN_REJECT_FLOW_H_
+#define HEDERA_SDK_CPP_TOKEN_REJECT_FLOW_H_
+
+#include "AccountId.h"
+#include "NftId.h"
+#include "TokenId.h"
+#include "TransactionResponse.h"
+
+namespace Hedera
+{
+class Client;
+class PrivateKey;
+}
+
+namespace Hedera
+{
+/**
+ * Reject undesired token(s) and dissociate in a single flow.
+ */
+class TokenRejectFlow
+{
+public:
+  /**
+   * Execute the Transactions in this flow (TokenRejectTransaction and TokenDissociateTransaction).
+   *
+   * @param client The Client to use to submit these Transactions.
+   * @return The TransactionResponse object of the flow.
+   * @throws MaxAttemptsExceededException If this Executable attempts to execute past the number of allowable attempts.
+   * @throws PrecheckStatusException      If this Executable fails its pre-check.
+   * @throws UninitializedException       If the input Client has not yet been initialized.
+   */
+  TransactionResponse execute(const Client& client);
+
+  /**
+   * Execute the Transactions in this flow (TokenRejectTransaction and TokenDissociateTransaction) with timeout.
+   *
+   * @param client  The Client to use to submit these Transactions.
+   * @param timeout The desired timeout for the execution of these Transactions.
+   * @return The TransactionResponse object of the flow.
+   * @throws MaxAttemptsExceededException If this Executable attempts to execute past the number of allowable attempts.
+   * @throws PrecheckStatusException      If this Executable fails its pre-check.
+   * @throws UninitializedException       If the input Client has not yet been initialized.
+   */
+  TransactionResponse execute(const Client& client, const std::chrono::system_clock::duration& timeout);
+
+  /**
+   * Freeze the TokenRejectTransaction with a Client. The Client's operator will be used to generate a transaction
+   * ID, and the client's network will be used to generate a list of node account IDs.
+   *
+   * NOTE: Since Client's can't be copied, this TokenRejectFlow will store a pointer to the input Client. It is the
+   * responsibility of the user to make sure that the Client does not go out of scope or get destroyed until this
+   * TokenRejectFlow is done executing, otherwise this will crash upon execution.
+   *
+   * @param client The Client with which to freeze the TokenRejectTransaction.
+   * @return A reference to this TokenRejectFlow object with the newly-set freeze Client.
+   * @throws UninitializedException If Client operator has not been initialized.
+   */
+  TokenRejectFlow& freezeWith(const Client& client);
```

Comment:
I think it makes sense to just pass in `const Client* client` here, since it's only grabbing its pointer and also that would conform to how `Transaction::freezeWith` does it